src/Symfony/Component/Workflow/Workflow.php

```diff
@@ -309,13 +315,13 @@ private function buildTransitionBlockerListForTransition(object $subject, Markin
         return new TransitionBlockerList();
     }
 
-    private function guardTransition(object $subject, Marking $marking, Transition $transition): ?GuardEvent
+    private function guardTransition(object $subject, Marking $marking, Transition $transition, array $context = []): ?GuardEvent
     {
         if (null === $this->dispatcher) {
             return null;
         }
 
-        $event = new GuardEvent($subject, $marking, $transition, $this);
+        $event = new GuardEvent($subject, $marking, $transition, $this, $context);
```

Comment:
For the guard event, I always refused to pass the workflow. Because people will use it to determine if the transition is enabled or not and this is wrong

Reply:
okay.
And you think they'll do the same with the context ?

Reply:
Oups, I made a typo in my comment. I meant context and not workflow. Sorry